bokeh/server/application_context.py

```diff
@@ -15,8 +15,9 @@ class ApplicationContext(object):
         data specific to an "instance" of the application.
     '''
 
-    def __init__(self, application):
+    def __init__(self, application, io_loop=None):
```

Comment:
I think application context is part of server implementation so we could require an io loop positional param